backend/generator.js

```diff
@@ -9,41 +9,114 @@ exports.executeScript = function (socket, formData) {
     socket.emit('err-logs', "Failed to generated documentation due to an error in input fields");
     return false;
   }
-  var email = formData.email;
-  var gitUrl = formData.gitUrl;
-  var docTheme = formData.docTheme;
-  var uniqueId = uuidV4();
-  var webUI = "true"
-
-  var donePercent = 0;
-
-  const args = [
-    "-g", gitUrl,
-    "-t", docTheme,
-    "-m", email,
-    "-u", uniqueId,
-    "-w", webUI
-  ];
-
-  var process = spawn("./generate.sh", args);
-
-  process.stdout.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('logs', {donePercent: (donePercent = donePercent + 4), data: data.toString()});
-  });
-
-  process.stderr.on('data', function (data) {
-    console.log(data.toString());
-    socket.emit('err-logs', data.toString());
-  });
-
-  process.on('exit', function (code) {
-    console.log('child process exited with code ' + code);
-    if (code === 0) {
-      socket.emit('success', {email: email, uniqueId: uniqueId, gitUrl: gitUrl});
-    } else {
-      socket.emit('failure', {errorCode: code});
-    }
-  });
-  return true;
+  // logic for one Repository
```

Comment:
What i did when i was working on implementing multiple projects' script was, I added an if statement in the `generate_ci.sh` that checked if the SUBPROJECT_URLS is set or not. If it was set, it executed another script, otherwise it executed `generate.sh`